cli/src/main.rs

```diff
@@ -0,0 +1,303 @@
+            Command::new("tes")
+                .about("TES subcommands")
+                .subcommand_required(true)
+                .arg_required_else_help(true)
+                .subcommand(
+                    Command::new("create")
+                        .about("Create a task")
+                        .arg(arg!(<TASK_FILE> "The task file to create"))
+                        // .arg(arg!(--url <URL> "The URL for the task"))
+                        .arg_required_else_help(true),
+                )
+
+                .subcommand(
+                    Command::new("list")
+                        .about("list all tasks")
+                        .arg(arg!(<params> "The parameters to get back"))
+                        .arg_required_else_help(true),
+                )
+                .subcommand(
+                    Command::new("get")
+                        .about("get task data")
+                        .arg(arg!(<id> "The id of the task which should be returned"))
+                        .arg(arg!(<view> "The view in which the task should be returned"))
+                        .arg_required_else_help(true),
+                )
+                .subcommand(
+                    Command::new("status")
+                        .about("get status of the task")
+                        .arg(arg!(<id> "The id of the task which should be returned"))
+                        .arg_required_else_help(true),
+                )
+                .subcommand(
+                    Command::new("cancel")
+                        .about("cancel the task")
+                        .arg(arg!(<id> "The id of the task which should be cancel"))
+                        .arg_required_else_help(true),
+                ),
+        );
+
+    let matches = cmd.clone().get_matches();
+
+    match matches.subcommand() {
+        Some(("tes", sub)) => {
+            if let Some(("create", sub)) = sub.subcommand() {
+                let task_file = sub.value_of("TASK_FILE").unwrap();
+                // let url = sub.value_of("url").unwrap();
+                let path = Path::new(task_file);
+                if !path.exists() {
+                    eprintln!("File does not exist: {:?}", path);
+                }
+                let task_json = match fs::read_to_string(path) {
+                    Ok(contents) => contents,
+                    Err(e) => {
+                        eprintln!("Failed to read file: {}", e);
+                        task_file.to_string()
+                    },
+                };
+                let testask: TesTask = serde_json::from_str(&task_json)
+                    .map_err(|e| format!("Failed to parse JSON: {}", e))?;
+                let mut config = load_configuration();
+                if config.base_path == "localhost" {
+                    let funnel_url = ensure_funnel_running().await;
+                    config.set_base_path(&funnel_url);
+                }
+                match TES::new(&config).await {
```

Comment:
@aaravm, I agree that a TES instance could be created in the beginning of `Some(("tes", sub)) => {` section, same with config load. It will significantly simplify this code

Reply:
I have refactored the code